### ISSUE TITLE
TestJcmd testMatchDisplayNamePartially Expected string not found

### ISSUE THREAD

**Issue by pshipton** [eclipse-openj9/openj9#19165]
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.functional_x86-64_windows_Release_testList_0/86
TestJcmd_0 
```
20:40:58  FAILED: testMatchDisplayNamePartially
20:40:58  java.lang.AssertionError: Expected string not found: java.util.HashMap expected [true] but found [false]
20:40:58  	at org.testng.Assert.fail(Assert.java:96)
20:40:58  	at org.testng.Assert.failNotEquals(Assert.java:776)
20:40:58  	at org.testng.Assert.assertTrue(Assert.java:44)
20:40:58  	at org.openj9.test.attachAPI.TestJcmd.testDisplayNameHelper(TestJcmd.java:372)
20:40:58  	at org.openj9.test.attachAPI.TestJcmd.testMatchDisplayNamePartially(TestJcmd.java:383)
```

**Comment by pshipton** [comment]
https://openj9-jenkins.osuosl.org/job/Test_openjdk17_j9_extended.functional_x86-64_windows_Nightly_testList_0/680
TestJcmd_0
```
23:34:00  FAILED: testMatchDisplayNamePartially
23:34:00  java.lang.AssertionError: Expected string not found: java.util.HashMap expected [true] but found [false]
23:34:00  	at org.testng.Assert.fail(Assert.java:96)
23:34:00  	at org.testng.Assert.failNotEquals(Assert.java:776)
23:34:00  	at org.testng.Assert.assertTrue(Assert.java:44)
23:34:00  	at org.openj9.test.attachAPI.TestJcmd.testDisplayNameHelper(TestJcmd.java:372)
23:34:00  	at org.openj9.test.attachAPI.TestJcmd.testMatchDisplayNamePartially(TestJcmd.java:383)
```

**Comment by pshipton** [comment]
@JasonFengJ9 I assume this is caused by the following, can you pls take a look
https://github.com/eclipse-openj9/openj9/pull/19069
https://github.com/eclipse-openj9/openj9/pull/19089

**Comment by JasonFengJ9** [comment]
JDK11 50x grinder - https://openj9-jenkins.osuosl.org/job/Grinder/3384/
JDK17 50x grinder - https://openj9-jenkins.osuosl.org/job/Grinder/3385/

**Comment by JasonFengJ9** [comment]
The test in question `TestJcmd.java` invokes `TargetManager` to create a target VM, then starts another process for `Jcmd` with corresponding arguments.
https://github.com/eclipse-openj9/openj9/blob/f2c434f21268841f3a42d2766d0556fdc35e0183/test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJcmd.java#L357-L367
From the AttachAPI log files, the target VM was started successfully but exits promptly w/o responding an attaching request. The instrumental log message from the target VM didn't appear in the test console output, will add tracepoints to understand why it didn't wait for the attaching request.
This seems a test issue.

**Comment by JasonFengJ9** [comment]
The attaching failed due to a `SocketException`:
```
java.net.SocketException: Software caused connection abort: socket write error
        at java.base/java.net.SocketOutputStream.socketWrite0(Native Method)
        at java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)
        at java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:129)
        at java.base/openj9.internal.tools.attach.target.AttachmentConnection.streamSend(AttachmentConnection.java:105)
        at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.executeDiagnosticCommand(OpenJ9VirtualMachine.java:318)
        at jdk.jcmd/openj9.tools.attach.diagnostics.attacher.AttacherDiagnosticsProvider.executeDiagnosticCommand(AttacherDiagnosticsProvider.java:55)
        at jdk.jcmd/openj9.tools.attach.diagnostics.tools.Util.runCommandAndPrintResult(Util.java:78)
```
This error in question is https://learn.microsoft.com/en-us/previous-versions/ms832256(v=msdn.10)?redirectedfrom=MSDN

Re-trying the test (3 times) passed consistently in 200x grinder.
`Jcmd` might be improved with re-tries in case this `SocketException`.
The failing test is a new test added recently via
* https://github.com/eclipse-openj9/openj9/pull/19069 

This behaviour is not a regression and will be addressed in 0.46.
